File: 0624-maximum-distance-in-arrays/0624-maximum-distance-in-arrays.cpp
class Solution {
public:
    int maxDistance(vector<vector<int>>& arrays) {
        int eMin = 1e5, eMax = -1e5;
        int diff = 0;
        for (vector<int>& arr: arrays) {
            diff = max({diff, eMax - arr[0], arr.back() - eMin});
            eMin = min(eMin, arr[0]);
            eMax = max(eMax, arr.back());
        }
        return diff;
    }
};

File: 0624-maximum-distance-in-arrays/0624-maximum-distance-in-arrays.kt
class Solution {
    fun maxDistance(arrays: List<List<Int>>): Int {
        var minEle : Int = 1e5.toInt();
        var maxEle : Int = (-1e5).toInt();
        var diff : Int = 0;

        for (arr in arrays) {
            diff = max(diff, max(arr.last() - minEle, maxEle - arr[0]));
            minEle = min(minEle, arr[0]);
            maxEle = max(maxEle, arr.last());
        }

        return diff;
    }
}